gin.go

```diff
@@ -281,6 +281,11 @@ func iterate(path, method string, routes RoutesInfo, root *node) RoutesInfo {
 	return routes
 }
 
+// ResetRoutes reset routes in Engine, used when application need to change routes dynamicly
+func (engine *Engine) ResetRoutes() {
+	engine.trees = make(methodTrees, 0, 9)
+}
+
```

Comment:
If the point of this is to change the router while it's running. Is this thread safe?